src/pages/ItemManagePage.js

```diff
@@ -0,0 +1,288 @@
+      newObjectValues = { vvmStatus: false };
+      // Account for 0 & NaN (From entering a non-numeric character)
+    } else if (parsedSplitValue) {
+      const newBatchValues = { totalQuantity: parsedSplitValue, vvmStatus: false };
+      data.push(createRowObject(currentBatch, newBatchValues));
+      newObjectValues = { vvmStatus: true, totalQuantity: totalQuantity - parsedSplitValue };
+    }
+    this.updateObject(newObjectValues, { isModalOpen: false });
+  };
+
+  onApplyChanges = () => {
+    // TODO:
+    // Confirmation dialog
+    // Create inventory adjustments for any failed vvm status'
+    // Create repacks for any location changes
+    // Pop this page off the navigation stack.
+    // If more data is needed to create these repacks and inventory
+    // adjustments, they can easily be added to the row object with
+    // no side-effects. e.g. the ItemBatch itself or just
+    // some extra fields.
+  };
+
+  onDispose = () => {
+    // TODO:
+    // Open a modal to select the doses?
+    // Set the row to have some distinctive marker to show it
+    // is being disposed of.
+    // Set the dispose icon to one which will 'undelete' this row.
+  };
+
+  // Called on selecting a fridge in the fridge selection modal,
+  // just update the location of the currentBatch and close the modal.
+  onFridgeSelection = location => {
+    this.updateObject({ location }, { isModalOpen: false });
+  };
+
+  // Called on toggle the vvm toggle bar. If being set to PASS, will just update
+  // the object held in the closure to a PASS VVM status. Otherwise, opens a
+  // modal for a user to enter the doses affected and set the currentBatch.
+  onVvmToggle = currentBatch => ({ newState }) => {
+    if (!newState) return this.setState({ modalKey: 'vvmStatus', isModalOpen: true, currentBatch });
+    return this.setState({ currentBatch }, () => this.updateObject({ vvmStatus: true }));
+  };
+
+  // Method which controls all modals. Will open a modal corresponding to the
+  // modal key in renderModal and set the passed itemBatch as the current
+  // itemBatch. If called with no parameters, will clear the currentBatch
+  // and close the modal.
+  onModalUpdate = ({ modalKey, itemBatch } = {}) => () => {
+    if (modalKey) this.setState({ modalKey, isModalOpen: true, currentBatch: itemBatch });
+    else this.setState({ isModalOpen: false, currentBatch: null });
+  };
+
+  /**
+   * RENDER HELPERS
+   */
+  renderCell = (key, itemBatch) => {
+    const { location, vvmStatus } = itemBatch;
+    const modalUpdateProps = { modalKey: key, itemBatch };
+    const hasFridges = this.FRIDGES && this.FRIDGES.length > 0;
+    const usingFridge = vvmStatus !== false && hasFridges;
+    switch (key) {
+      default:
+        return itemBatch[key];
+      case 'location':
+        return (
+          <IconCell
+            text={
+              (hasFridges && 'No Fridges') ||
+              (vvmStatus !== false ? location.description : 'Discarded')
```

Comment:
can check for location being null here instead of in createRowObject

Reply:
and of course change to: `Unassigned`

Reply:
cf6b43bbf60f0d363d4a2ca12e6416565d756ad9 - Went with "Unnamed?" seems more.. right?

Reply:
@joshxg i meant if location is null then 'Unassigned', we will enforce fridge description to be entered and being unique for a store.

Reply:
1b0cdefb68133463cad34740081999d16d504ffb